server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,321 @@
+                    registerDecommissionAttribute(decommissionAttribute, listener);
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.debug(new ParameterizedMessage(
+                        "failure in removing decommissioned cluster manager eligible nodes from voting config"), exp);
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public AddVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new AddVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    private void clearVotingConfigAfterSuccessfulDecommission() {
+        final ClearVotingConfigExclusionsRequest clearVotingConfigExclusionsRequest = new ClearVotingConfigExclusionsRequest();
+        clearVotingConfigExclusionsRequest.setWaitForRemoval(true);
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClearVotingConfigExclusionsAction.NAME,
+            clearVotingConfigExclusionsRequest,
+            new TransportResponseHandler<ClearVotingConfigExclusionsResponse>() {
+                @Override
+                public void handleResponse(ClearVotingConfigExclusionsResponse response) {
+                    logger.info("successfully cleared voting config after decommissioning");
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.debug(new ParameterizedMessage(
+                        "failure in clearing voting config exclusion after decommissioning"), exp);
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public ClearVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new ClearVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    /**
+     * Registers new decommissioned attribute metadata in the cluster state
+     * <p>
+     * This method can be only called on the cluster-manager node. It tries to create a new decommissioned attribute on the master
+     * and if it was successful it adds new decommissioned attribute to cluster metadata.
+     * <p>
+     * This method should only be called once the eligible cluster manager node having decommissioned attribute is abdicated
+     *
+     * @param decommissionAttribute register decommission attribute in the metadata request
+     * @param listener              register decommission listener
+     */
+    private void registerDecommissionAttribute(
```

Comment:
Can we delegate private method to the present helper, can we avoid static methods, not a great fan

Reply:
Moved independent methods to controller